Player can access the profile menu

PhoebeMitchell opened the pull request:
## Summary
<!--- Describe the change below, including rationale and design decisions -->
Updates the evidence menu to support showing profiles. Actor dictionaries now use the same system as evidence (dictionary created on a monobehaviour instead of in the scriptable object). The logic from EvidenceDictionary is now in its own class called ObjectDictionary which the ActorDictionary and EvidenceDictionary monobehaviours use. ActorData and Evidence now implement ICourtRecordObject which allows their names, icons, and descriptions to be accessed. Also features tests for the evidence menu and ObjectDictionary class.

## Before/after screenshots and/or animated gif
<!--- Skip this if not applicable -->
![profilesdemo2](https://user-images.githubusercontent.com/67964179/132934810-b9e0b182-6462-4ef6-8edf-f42d955261bf.gif)

## Testing instructions
<!--- What steps can be taken to manually verify the changes? -->
Open a scene with the StandardInclusion asset and enter play mode. Press z to open the evidence menu, then press c to open the profiles menu. Navigate with the arrow keys (and check for spelling mistakes in each bio).
Open the unity test runner and run the play mode and edit mode tests.
The screenshots below show the tests passing.
<img width="564" alt="Screenshot 2021-09-16 at 14 30 48" src="https://user-images.githubusercontent.com/67964179/133621427-878c360c-a616-4be8-abfb-f081994640df.png">
<img width="565" alt="Screenshot 2021-09-16 at 14 30 55" src="https://user-images.githubusercontent.com/67964179/133621433-399d8f7d-b0a6-4863-823f-9e8268b1f952.png">

## Additional information
<!--- Check any relevant boxes with "x" -->
- `[x]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource: 
  <!--- Include any project card, github issue, etc. associated with this PR -->
#11

TooManyMoths committed:
Abstracted EvidenceDictionary

Its now uses a dictionary wrapper that takes Objects

TooManyMoths committed:
Merge branch 'develop' into Player-can-access-the-profile-menu

TooManyMoths committed:
Evidence and Actors now use the same dictionary system

TooManyMoths committed:
Moved some classes to their correct directories

TooManyMoths committed:
Evidence menu can be toggled to profile menu without any visual bugs

TooManyMoths committed:
Created ObjectDictionary tests

TooManyMoths committed:
Updated name of property to be more accurate. List -> Array

TooManyMoths committed:
Fixed one variable name that wasn't updated in previous commit

TooManyMoths committed:
Began work on evidence menu test.

TooManyMoths committed:
Merge branch 'develop' into Player-can-access-the-profile-menu

TooManyMoths committed:
Created evidence menu play mode tests

TooManyMoths committed:
Evidence menu mouse tests

TooManyMoths committed:
Updated standard inclusion

TooManyMoths committed:
Updated tests to reflect change to evidence dictionary initial evidence

TooManyMoths committed:
Added checking menu information updates when highlighting items to tests

TooManyMoths committed:
Fixed wrong ordering in dictionary components

TooManyMoths committed:
Added missing descriptions to tests and updated some variable names to be more accurate.

TooManyMoths committed:
Added bios to actors

TooManyMoths committed:
Added actor ages

TooManyMoths committed:
Fixed broken test

JuiceGrape commented on Assets/Scripts/Actor/ActorDictionary.cs:
```diff
@@ -1,25 +1,12 @@
-
-[CreateAssetMenu(fileName = "New Actor Dictionary", menuName = "Actors/Actor Dictionary")]
-public class ActorDictionary : ScriptableObject
+public class ActorDictionary : ObjectDictionaryInterface<ActorData, ActorList>, ICourtRecordObjectDictionary
```
Kudos: Nice one! Sexy ass implementation, shit like this makes me happy

JuiceGrape commented on Assets/Scripts/Actor/ActorData.cs:
```diff
@@ -21,4 +33,24 @@ public class ActorData : ScriptableObject
+    private string GenerateNameWithAge()
+    {
+        string age = "???";
+        switch (_ageCertainty)
```
For extendability it is better to leave the default as an error and then handle all known cases in the switch. Also handling the default in the switch is better practice for this reason, so when a new enum gets added and they forget to extend this switch, it is instantly noticeable

ViMaSter replied:
Agreed! `AgeCertainty.Unknown` is a valid state and should be handled appropriately. The `default` should be treated as an exception or error.

---

I'd personally also suggest converting it to a [switch expression](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression) like so:
```cs
string age = _ageCertainty switch {
    AgeCertainty.Known => Age.ToString(),
    AgeCertainty.Guess => $"Maybe {Age}?",
    AgeCertainty.Unknown => "???",
    _ => throw new ArgumentOutOfRangeException()
};
```

PhoebeMitchell replied:
Changed to use ViMaSter's suggestion.

JuiceGrape commented on Assets/Scripts/Evidence/EvidenceDictionary.cs:
```diff
@@ -1,157 +1,22 @@
-using UnityEngine;
-
-public class EvidenceDictionary : MonoBehaviour
+public class EvidenceDictionary : ObjectDictionaryInterface<Evidence, EvidenceList>, ICourtRecordObjectDictionary
```
Kudos: New implementation looks much cleaner

JuiceGrape reviewed (dismissed):
Great work! Feel free to close the kudos comments when read

ViMaSter commented on Assets/Tests/PlayModeTests/Scripts/EvidenceMenu/EvidenceMenuMouseTests.cs:
```diff
@@ -0,0 +1,92 @@
+        /// Presses the navigation buttons multiple times then checks if
+        /// the first menu item is the same as the one that the menu thinks is selected.
+        /// </summary>
+        // [UnityTest, Order(1)]
```
This test appears to be commented out and doesn't show up on the Test Runner. I'd suggest either removing the code for it or integrating it somehow!

PhoebeMitchell replied:
Good catch! Commented out that method for debugging purposes and must have missed the attributes when uncommenting. Fixed.

ViMaSter commented on Assets/Tests/PlayModeTests/Scripts/EvidenceMenu/EvidenceMenuMouseTests.cs:
```diff
@@ -0,0 +1,92 @@
+using System.Linq;
+using NUnit.Framework;
+using Tests.PlayModeTests.Tools;
+using TMPro;
```
Nit: This namespace seems to be unused in this file. I'd suggest removing it.

PhoebeMitchell replied:
Done

ViMaSter commented on Assets/Tests/PlayModeTests/Scripts/EvidenceMenu/EvidenceMenuMouseTests.cs:
```diff
@@ -0,0 +1,92 @@
+            // Loop through the menu items and check if they highlight correctly
+            for (int i = 1; i < menuItems.Length; i++)
+            {
+                MenuItem menuItem = menuItems.First(menuItem => menuItem.gameObject.name == $"EvidenceMenuItem ({i})");
```
Nit: Having the outer variable (`MenuItem menuItem`) **and** parameter of the `First()`-predicate (.First(menuItem`) is a bit confusing to read for me and also prevents the variable from being used inside the scope. I'd suggest renaming either the inner or outer variable.

PhoebeMitchell replied:
Changed First() parameter to "item".

ViMaSter commented on Assets/Tests/PlayModeTests/Scripts/EvidenceMenu/EvidenceMenuKeyboardTests.cs:
```diff
@@ -0,0 +1,172 @@
+            Menu menu = _evidenceMenu.GetComponent<Menu>();
+            TextMeshProUGUI[] evidenceTextBoxes = _evidenceMenu.GetComponentsInChildren<TextMeshProUGUI>();
+            TextMeshProUGUI evidenceName =
+                evidenceTextBoxes.First(evidenceTextBoxe => evidenceTextBoxe.gameObject.name == "EvidenceName");
```
Nit: Typo `evidenceTextBoxe` -> `evidenceTextBox`.

PhoebeMitchell replied:
Fixed

ViMaSter commented on Assets/Tests/PlayModeTests/Scripts/EvidenceMenu/EvidenceMenuKeyboardTests.cs:
```diff
@@ -0,0 +1,172 @@
+            Menu menu = _evidenceMenu.GetComponent<Menu>();
+            TextMeshProUGUI[] evidenceTextBoxes = _evidenceMenu.GetComponentsInChildren<TextMeshProUGUI>();
+            TextMeshProUGUI evidenceName =
+                evidenceTextBoxes.First(evidenceTextBoxe => evidenceTextBoxe.gameObject.name == "EvidenceName");
```
Nit: Typo `evidenceTextBoxe` -> `evidenceTextBox`.

PhoebeMitchell replied:
Fixed

ViMaSter replied:
Just FYI: The latest version of this file still contains the typo. 👀
https://github.com/Studio-Lovelies/GG-JointJustice-Unity/blob/0bea8fa254d3b558d48947255e17a399e51e4f56/Assets/Tests/PlayModeTests/Scripts/EvidenceMenu/EvidenceMenuKeyboardTests.cs#L131

PhoebeMitchell replied:
Fixed. Must have come back when reverting back to humble object pattern.

ViMaSter commented on Assets/Tests/PlayModeTests/Tools/InputTestTools.cs:
```diff
@@ -0,0 +1,102 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.InputSystem.Controls;
+using UnityEngine.UI;
```
Nit: This namespace seems to be unused in this file. I'd suggest removing it.

PhoebeMitchell replied:
Done

ViMaSter commented on Assets/Tests/PlayModeTests/Tools/InputTestTools.cs:
```diff
@@ -0,0 +1,102 @@
+        /// <param name="seconds">The number of seconds to hold the key for each press.</param>
+        /// <param name="repeats">The number of times to press the key.</param>
+        /// <returns></returns>
+        public IEnumerator RepeatPressForSeconds(ButtonControl control, float seconds, int repeats)
```
Nit: `RepeatPressForSeconds()` seems to be unused. I'd personally suggest either removing it (if we end up needing it, we can add it later) or merging it with `PressForSeconds()` by adding `repeats` as an optional parameter (`public IEnumerator PressForSeconds(ButtonControl control, float seconds, int repeats = 1)`).  
I'm always a big fan of removing unused code, that way there's less of it to keep in mind when working with it!

PhoebeMitchell replied:
Removed "Repeat" methods. Modified PressForFrame and PressForSeconds to allow repeats.

ViMaSter commented on Assets/Tests/PlayModeTests/Tools/InputTestTools.cs:
```diff
@@ -0,0 +1,102 @@
+    /// Contains useful methods used when testing features that use Unity Input System.
+    /// Also contains input devices that should be used to pass ButtonControls to the methods.
+    /// </summary>
+    public class InputTestTools : InputTestFixture
```
Wrapping this functionality in a helper-class is a really solid addition! Always a big fan of the [rule of 3](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming))! 👍

ViMaSter commented on Assets/Tests/EditModeTests/ObjectDictionaryTests.cs:
```diff
@@ -0,0 +1,173 @@
+            for (int i = 0; i < count; i++)
+            {
+                list.Add(new T());
+            }
+            return list;
+        }
+        
+        /// <summary>
+        /// Creates a list of scriptable objects to use in above tests.
+        /// </summary>
+        /// <param name="count">The number of scriptable objects in the list.</param>
+        /// <typeparam name="T">The type of scriptable object to create a list of.</typeparam>
+        /// <returns>The list of scriptable objects created.</returns>
+        private List<T> CreateScriptableObjectList<T>(int count) where T : ScriptableObject
+        {
+            var list = new List<T>();
+            for (int i = 0; i < count; i++)
+            {
+                T scriptableObject = ScriptableObject.CreateInstance<T>();
+                scriptableObject.name = i.ToString();
+                list.Add(scriptableObject);
+            }
+            return list;
+        }
+
+        /// <summary>
+        /// Create a list of game objects and give them corresponding to the order they were created in.
+        /// </summary>
+        /// <param name="count">The number of game objects to create.</param>
+        /// <returns>The list of game objects created.</returns>
+        private List<GameObject> CreateGameObjectList(int count)
+        {
+            var gameObjectList = CreateObjectList<GameObject>(count);
+            for (int i = 0; i < gameObjectList.Count; i++)
+            {
+                gameObjectList[i].name = i.ToString();
+            }
+            return gameObjectList;
+        }
+    }
```
Nit: These methods can be made `static` and return `IEnumerable<>` instead of `List<>`.

PhoebeMitchell replied:
Done

ViMaSter commented on Assets/Tests/EditModeTests/ObjectDictionaryTests.cs:
```diff
@@ -0,0 +1,173 @@
+        /// <summary>
+        /// Test to check if values can be swapped with other values.
+        /// Swaps the first 50 objects with the remaining 50 objects in the dictionary
+        /// then checks if their names are correct.
+        /// </summary>
+        [Test]
+        public void SubstituteValueInDictionaryWithAlt()
+        {
+            int objectDictionaryCount = 100;
+            var objectDictionary = new ObjectDictionary<GameObject>(CreateGameObjectList(objectDictionaryCount), CreateGameObjectList(objectDictionaryCount));
+            for (int i = 0; i < objectDictionaryCount; i++)
+            {
+                objectDictionary.SubstituteValueWithAlt(i.ToString(), objectDictionary[(objectDictionary.Count - (i + 1)).ToString()]);
+            }
+
+            for (int i = 0; i < objectDictionaryCount / 2; i++)
+            { 
+                Assert.AreEqual((objectDictionaryCount - (i + 1)).ToString(), objectDictionary[i.ToString()].name);
+            }
+
+            for (int i = 50; i < objectDictionaryCount; i++)
+            {
+                Assert.AreEqual(i.ToString(), objectDictionary[i.ToString()].name);
+            }
+        }
+
+        /// <summary>
+        /// Test to check if objects can be retrieved from the dictionary by index.
+        /// In this case an object at an index will have the same name as the index number.
+        /// </summary>
+        [Test]
+        public void GetObjectFromDictionaryAtIndex()
+        {
+            int objectDictionaryCount = 1000;
+            var objectDictionary = new ObjectDictionary<GameObject>(CreateGameObjectList(objectDictionaryCount), CreateGameObjectList(objectDictionaryCount));
+            for (int i = 0; i < objectDictionaryCount; i++)
+            {
+                Assert.AreEqual(i.ToString(), objectDictionary[i].name);
+            }
+        }
```
Nit: The `*Count` variables and `numberOfObjectsToRemove` used during setup of each of these cases can be made `const`, to prevent them from being accidentally changed during runtime.

PhoebeMitchell replied:
Done

ViMaSter commented on Assets/Scripts/TextDecoder/DialogueController.cs:
```diff
@@ -166,30 +166,12 @@ public void SetMenuOpen(bool isMenuOpen)
     /// </summary>
     /// <param name="presentedObject">Name of the object you want to present to the court</param>
-    private void HandlePresenting(string presentedObject)
+    public void HandlePresenting(ICourtRecordObject presentedObject)
```
Really intuitive way of simplifying the public interface of this class while making it even less error-prone! 👍

ViMaSter commented on Assets/Scripts/ObjectDictionary/ObjectDictionaryInterface.cs:
```diff
@@ -0,0 +1,65 @@
+/// This class allows interaction between unity objects and ObjectDictionary objects.
+/// Allows assigning of an object with an IObjectList interface to convert to a dictionary.
+/// </summary>
+/// <typeparam name="T">The type of object to store in the dictionary.</typeparam>
```
Nit: I personally find specifying `The type of values in the dictionary.` to make it easier to read, as only the value type is being set.

PhoebeMitchell replied:
Changed

ViMaSter commented on Assets/Scripts/ObjectDictionary/ObjectDictionaryInterface.cs:
```diff
@@ -0,0 +1,65 @@
+/// <typeparam name="S">The type that the assignable IObjectList should use.</typeparam>
+public abstract class ObjectDictionaryInterface<T, S> : MonoBehaviour where T : Object where S : IObjectList<T>
+{
+    [SerializeReference, Tooltip("Drag an Actor Dictionary here containing all the actors available in the scene.")]
+    private S _masterObjectList;
+    
+    [field: SerializeField, Tooltip("Add any evidence that the player should have at the start of the scene here.")]
+    protected T[] CurrentObjectList { get; private set; }
```
Nit: The tooltips speak of `actors` and `evidence` explicitly, while the fields are technically being abstract. I'd change this to `ICourtRecordObject` or `Object` instead.

PhoebeMitchell replied:
Remnants of old implementations. Updated.

ViMaSter commented on Assets/Scripts/ObjectDictionary/ObjectDictionaryInterface.cs:
```diff
@@ -0,0 +1,65 @@
+public abstract class ObjectDictionaryInterface<T, S> : MonoBehaviour where T : Object where S : IObjectList<T>
+{
+    [SerializeReference, Tooltip("Drag an Actor Dictionary here containing all the actors available in the scene.")]
+    private S _masterObjectList;
```
Nit: The term `master` is immediately associated with a `master/slave`-relationship (think SATA, IDE, databases), which doesn't really apply here. Similarly to the tooltip-text, I'd prefer `availableObjectList`, as it contains a list of all potential objects that can be added to the `CurrentObjectList`.

PhoebeMitchell replied:
Changed to _availableObjectList

ViMaSter commented on Assets/Scripts/ObjectDictionary/ObjectDictionaryInterface.cs:
```diff
@@ -0,0 +1,65 @@
+    /// </summary>
+    /// <param name="index">The index of the object to get.</param>
+    /// <returns>The object at the specified index.</returns>
+    public T GetObjectAtIndex(int index)
```
Nit: This method is never used. `GetObjectAtIndex()` is only ever used on `ICourtRecordObjectDictionary`-objects, which - as it's an interface - require an explicit implementation and - as it exists as a shallower type - can't fall back to this base implementation. I'd personally suggest removing it.

PhoebeMitchell replied:
Removed.

ViMaSter commented on Assets/Scripts/ObjectDictionary/IObjectList.cs:
```diff
@@ -0,0 +1,6 @@
+using UnityEngine;
+
+public interface IObjectList<T> where T : Object
```
Nit: I'd suggest using `<out T>` instead of `<T>` here, which would make this [a covariant](https://docs.microsoft.com/en-us/dotnet/standard/generics/covariance-and-contravariance#generic-interfaces-with-covariant-type-parameters), allowing it to be used more broadly.

PhoebeMitchell replied:
Done

ViMaSter commented on Assets/Scripts/ObjectDictionary/ICourtRecordObject.cs:
```diff
@@ -0,0 +1,10 @@
+
+public interface ICourtRecordObject
+{ 
+    public string name { get; }
```
Nit: This might be something Unity-specific related to `name` I'm unaware of, but all other properties in this interface start with an uppercase letter, so I'd personally prefer aligning them to all use UpperCamelCase.

PhoebeMitchell replied:
This is to allow the name property of Object to be accessed. Would you suggest changing it to Name and then doing `string Name => name;` on classes that implement this instead?

ViMaSter replied:
That actually sounds pretty good!  
The [official docs of `UnityEngine.Object.name`](https://docs.unity3d.com/ScriptReference/Object.html) state that `Components share the same name with the game object and all attached components`. Given that, I'd just go a step further and change it to `DisplayName`. That way this property is distinctly tied to our game and - if we choose to do so - we can use the value from the engine by using `=> name` on any class that derived from `UnityEngine.Object`.

PhoebeMitchell replied:
Changed to InstanceName

ViMaSter commented on Assets/Scripts/Evidence/EvidenceMenu.cs:
```diff
@@ -179,4 +220,29 @@ private bool HasEvidenceDictionary()
+        UpdateEvidenceMenu();
+        _menu.SelectInitialButton();
+    }
 }
```
Nit: Strictly not part of this PR, but the `System` namespace in this file seems to be unused. I'd suggest removing it.

PhoebeMitchell replied:
Done

ViMaSter commented on Assets/Scripts/ObjectDictionary/ObjectDictionary.cs:
```diff
@@ -0,0 +1,185 @@
+    /// </summary>
+    /// <param name="array">The list to convert to a dictionary.</param>
+    /// <returns>The dictionary that the list has been converted to.</returns>
+    private Dictionary<string, T> ArrayToDictionary(IEnumerable<T> array)
```
Nit: I'd recommend making this method `static` as it doesn't appear to be using any instance members.

PhoebeMitchell replied:
Done

ViMaSter reviewed (changes requested):
Really solid implementation! Always great to see a bunch of tests being added as well! <3

My only required changes (anything that's not prefixed with `Nit:` or got a 👍) would be:
- the typos ([1](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/pull/52/files#r706865549), [2](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/pull/52/files#r706865542)) and
- [the test case that's commented out](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/pull/52/files#r706865061)

---

A sidenote:

I noticed, that the class `ObjectDictionary` does not implement `IDictionary` and the `ObjectDictionaryInterface` is actually an `abstract class` instead of an `interface`.  
This made this change a bit difficult to read for me initially.  

Looking through the source it definitely is possible to get an understanding of what these classes actually do, but as the terms `interface` and `dictionary` are already being used for definitions inside the language-standard, my mind immediately jumped to a set of behaviors and available methods that aren't actually present inside these classes when initially looking at these classes, which might cause confusion for people working on it the first time or after a long period of time has passed.

If I can offer two suggestions:
1. From what I can see, `ObjectDictionary` is only ever used inside `ObjectDictionaryInterface` and it's test suite. I'd recommend the following approach:
   * Integrate the logic of `ObjectDictionary` into `ObjectDictionaryInterface` making it one class
   * Rename this new `ObjectDictionaryInterface` class to `ObjectDictionary`
   This will result in less nesting of classes, while also reducing potential confusion caused by the word `Interface`
2. Instead of creating and implementing the custom interface `IObjectList`, implement `IDictionary<>` directly
   This will result in associations with the word `Dictionary` actually being fulfilled
   _Note: The same also applies to `ICourtRecordObjectDictionary`._

I'd be happy to hear your thoughts on that; as it's a bigger topic I'm also open to have a quick voice chat about it! However, as the functionallity operates just fine, this is just a suggestion and not blocking approval!

---

I really appreciate you adhering to the boyscout principle of "leaving the campsite cleaner than you found it" and taking the time to create the `InputTestTools` class! This should keep flakey tests to an absolute minimum!

TooManyMoths committed:
Implemented changes requested on PR

TooManyMoths committed:
Moved ObjectDictionary logic to ObjectDictionaryInterface

TooManyMoths committed:
Updated ObjectDictionaryTests to reflect changes to ObjectDictionary

TooManyMoths committed:
Revert "Updated ObjectDictionaryTests to reflect changes to ObjectDictionary"

This reverts commit c987041739f134e5a6c3f827c12dc6756bbbc8c9.

TooManyMoths committed:
Revert "Moved ObjectDictionary logic to ObjectDictionaryInterface"

This reverts commit e43ce78c554f8dfec63d6ea1b0a85b5162542f1f.

ViMaSter commented on Assets/Tests/EditModeTests/ObjectDictionaryTests.cs:
```diff
@@ -0,0 +1,102 @@
+    {
+        private EvidenceDictionary _evidenceDictionary;
+
+        private string[] _evidenceNames =
```
Nit: This field can be made `readonly`.

PhoebeMitchell replied:
File no longer exists.

ViMaSter commented on Assets/Tests/EditModeTests/ObjectDictionaryTests.cs:
```diff
@@ -0,0 +1,102 @@
+using System.Collections.Generic;
+using System.Linq;
```
Nit: No symbols from these namespaces are used in the this file, so the first two lines of this file can safely be removed.

PhoebeMitchell replied:
File no longer exists.

ViMaSter commented on Assets/Scripts/ObjectDictionary/ObjectDictionary.cs:
```diff
@@ -0,0 +1,197 @@
+    /// <param name="objectName">The name of the object to look for in the dictionary.</param>
+    /// <param name="dictionaryToCheck">The dictionary to look for the object in.</param>
+    /// <returns>True if the object is in the dictionary, and false if it is not.</returns>
+    private bool IsObjectInDictionary(string objectName, Dictionary<string, T> dictionaryToCheck)
```
Nit: This method can be made static.

PhoebeMitchell replied:
File no longer exists. The version of this method in latest version is public. I'm told static methods are bad in OOP. Thoughts?

JuiceGrape replied:
Static methods are fine in OOP if you have a reason for making them. In this case, making this method static is a good idea if we want to use it in other classes as well, to avoid duplicate code. If we don't, leave it as is

PhoebeMitchell replied:
Made IsObjectInDictionary and EnumerableToDictionary methods static as their output is always the same if the input doesn't change.

ViMaSter commented on Assets/Scripts/Evidence/EvidenceDictionary.cs:
```diff
@@ -1,157 +1,24 @@
-using System;
-using System.Collections.Generic;
-using System.Linq;
 using UnityEngine;
```
Nit: No symbols from this namespace are used in the this file, so this `using`-directive can safely be removed too.

PhoebeMitchell replied:
File no longer exists.

ViMaSter commented on Assets/Scripts/Actor/ActorList.cs:
```diff
@@ -0,0 +1,10 @@
+using System;
+using System.Collections.Generic;
```
Nit: No symbols from this namespace are used in the this file, so this `using`-directive can safely be removed.

PhoebeMitchell replied:
Outdated.

ViMaSter commented on Assets/Scripts/Actor/ActorData.cs:
```diff
@@ -1,9 +1,16 @@
+using System;
 using System.Collections.Generic;
```
Nit: No symbols from this namespace are used in the this file, so this using-directive can safely be removed.

PhoebeMitchell replied:
Removed.

TooManyMoths committed:
Changed ObjectDictionary to be a wrapper for OrderedDictionary and renamed to OrderedObjectStorage

TooManyMoths committed:
Renamed testevidence folder to Test Evidence to fit naming style

TooManyMoths committed:
Merge branch 'develop' into Player-can-access-the-profile-menu

TooManyMoths committed:
Fixed basecanvas not being loaded in standardinclusion

TooManyMoths committed:
Merge branch 'develop' into Player-can-access-the-profile-menu

TooManyMoths committed:
Updated build.yml to say Windows-x86 instead of Windows-x68

ViMaSter commented on Assets/Scripts/Actor/ActorController.cs:
```diff
@@ -3,14 +3,16 @@
 using System.Collections.Generic;
```
Nit: No symbols are used from `using System;` and `using System.Collections;`, so the first two lines of this file can safely be removed.

PhoebeMitchell replied:
Will leave them to avoid changing too much of a file that isn't explicitly introduced in this PR.

ViMaSter commented on Assets/Scripts/Actor/ActorData.cs:
```diff
@@ -13,6 +20,12 @@ public class ActorData : ScriptableObject
     [field: SerializeField, Tooltip("Name displayed when referring to this actor or when they are speaking.")]
```
Nit: The property `DefaultSprite` isn't used (anymore?), could this be removed until functionality that uses it exists or would this result in a lot of Prefabs having to be updated?

PhoebeMitchell replied:
Removed as its intended functionality did not work it is and no longer used.

ViMaSter commented on Assets/Scripts/Evidence/Evidence.cs:
```diff
@@ -3,11 +3,11 @@
 using UnityEngine;
```
Nit: No symbols are used from `using System;` and `using System.Collections;`, so the first two lines of this file can safely be removed.

PhoebeMitchell replied:
Removed

ViMaSter commented on Assets/Scripts/Evidence/EvidenceController.cs:
```diff
@@ -2,6 +2,7 @@
 using System.Collections.Generic;
```
Nit: No symbols are used from `using System;` and `using System.Collections;`, so the first two lines of this file can safely be removed.

PhoebeMitchell replied:
Will leave them to avoid changing too much of a file that isn't explicitly introduced in this PR.

ViMaSter commented on Assets/Scripts/ObjectDictionary/ObjectInventory.cs:
```diff
@@ -0,0 +1,68 @@
+using System.Collections;
```
Nit: No symbols from this namespace are used in the this file, so this using-directive can safely be removed.

PhoebeMitchell replied:
Removed

ViMaSter reviewed (dismissed):
Looks good to me! Just a few nits, but nothing that's worth blocking in my opinion! Oh and thanks for integrating the typo fix for the build artifacts too.

TooManyMoths committed:
Implemented nitpicks from ViMaSter's review

lyndsiWilliams commented on Assets/Scripts/Actor/ActorData.cs:
```diff
@@ -21,4 +30,21 @@ public class ActorData : ScriptableObject
+    public Sprite Icon => Profile;
+    public string Description => Bio;
+
+    private string GenerateNameWithAge()
```
This method is missing a summary ✨

PhoebeMitchell replied:
Added

lyndsiWilliams reviewed (dismissed):
Found just one thing! It looks like you've got some failing tests as well, but otherwise this looks great!

TooManyMoths committed:
Added summary requested by Amythia

TooManyMoths committed:
Changed IsObjectInDictionary and EnumerableToDictionary methods to static.

Changed OrderedObjectStorage property to ObjectStorage to prevent name conflicting with its type.

TooManyMoths committed:
Changed IsObjectInDictionary and EnumerableToDictionary methods to static.

Changed OrderedObjectStorage property to ObjectStorage to prevent name conflicting with its type.

TooManyMoths committed:
Merge branch 'develop' into Player-can-access-the-profile-menu

TooManyMoths committed:
Merge branch 'Player-can-access-the-profile-menu' of https://github.com/Studio-Lovelies/GG-JointJustice-Unity into Player-can-access-the-profile-menu

TooManyMoths committed:
Fixed conflicts and profile menu label showing in evidence menu

JuiceGrape reviewed (approved):
Solved bug looks good, ready to merge!

lyndsiWilliams reviewed (approved):
Looks great! ✨

PhoebeMitchell committed:
Merge branch 'develop' into Player-can-access-the-profile-menu
